js/packages/openinference-instrumentation-llama-index/src/utils.ts

```diff
@@ -56,14 +63,94 @@ function getExecContext(span: Span) {
   return execContext;
 }
 
+/**
+ * If execution results in an error, push the error to the span.
+ * @param span
+ * @param error
+ */
+function handleError(span: Span, error: Error | undefined) {
+  if (error) {
+    span.recordException(error);
+    span.setStatus({
+      code: SpanStatusCode.ERROR,
+      message: error.message,
+    });
+    span.end();
+  }
+}
+
+function documentAttributes(
+  output: llamaindex.NodeWithScore<llamaindex.Metadata>[],
+) {
+  const docs: Attributes = {};
+  output.forEach(({ node, score }, idx) => {
+    if (node instanceof llamaindex.TextNode) {
+      const prefix = `${SemanticConventions.RETRIEVAL_DOCUMENTS}.${idx}`;
+      docs[`${prefix}.${SemanticConventions.DOCUMENT_ID}`] = node.id_;
+      docs[`${prefix}.${SemanticConventions.DOCUMENT_SCORE}`] = score;
+      docs[`${prefix}.${SemanticConventions.DOCUMENT_CONTENT}`] =
+        node.getContent();
+      docs[`${prefix}.${SemanticConventions.DOCUMENT_METADATA}`] =
+        safelyJSONStringify(node.metadata) ?? undefined;
+    }
+  });
+  return docs;
+}
+
+function queryEmbeddingAttributes(input: string, output: number[]) {
+  return {
+    [`${SemanticConventions.EMBEDDING_EMBEDDINGS}.0.${SemanticConventions.EMBEDDING_TEXT}`]:
+      input,
+    [`${SemanticConventions.EMBEDDING_EMBEDDINGS}.0.${SemanticConventions.EMBEDDING_VECTOR}`]:
+      output,
+  };
+}
+
+function textEmbeddingsAttributes(input: string[], output: number[][]) {
+  const embedAttr: Attributes = {};
+  input.forEach((str, idx) => {
+    embedAttr[
+      `${SemanticConventions.EMBEDDING_EMBEDDINGS}.${idx}.${SemanticConventions.EMBEDDING_TEXT}`
+    ] = str;
+  });
+
+  output.forEach((vector, idx) => {
+    embedAttr[
+      `${SemanticConventions.EMBEDDING_EMBEDDINGS}.${idx}.${SemanticConventions.EMBEDDING_VECTOR}`
+    ] = vector;
+  });
+  return embedAttr;
+}
+
+function hasId(obj: unknown): obj is ObjectWithID {
+  const objectWithIDMaybe = obj as ObjectWithID;
+  return "id" in objectWithIDMaybe && typeof objectWithIDMaybe.id === "string";
+}
+
+function hasModel(obj: unknown): obj is ObjectWithModel {
+  const ObjectWithModelMaybe = obj as ObjectWithModel;
+  return (
+    "model" in ObjectWithModelMaybe &&
+    typeof ObjectWithModelMaybe.model === "string"
+  );
+}
+
+function getModelName(obj: unknown) {
+  if (hasModel(obj)) {
+    return obj.model;
+  } else if (hasId(obj)) {
+    return obj.id;
+  }
+}
+
 export function patchQueryMethod(
-  original: typeof module.RetrieverQueryEngine.prototype.query,
+  original: QueryEngineQueryMethod,
```

Comment:
might rename this function / type to include the word retriever in there - e.g., RetrieverQueryEngineQueryMethod / pathcRetreiverQueryMethod